src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject), Name = "MPSCNNConvolutionDataSource")]
+	interface MPSCnnConvolutionDataSource
+	{
+		[Abstract]
+		[Export ("dataType")]
+		MPSDataType DataType { get; }
+
+		[Abstract]
+		[Export ("descriptor")]
+		MPSCnnConvolutionDescriptor Descriptor { get; }
+
+		[Abstract]
+		[Export ("weights")]
+		IntPtr Weights { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("biasTerms")]
+		IntPtr /* float* */ BiasTerms { get; }
+
+		[Abstract]
+		[Export ("load")]
+		bool Load { get; }
+
+		[Abstract]
+		[Export ("purge")]
+		void Purge ();
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+
+		[Export ("rangesForUInt8Kernel")]
+		IntPtr RangesForUInt8Kernel { get; }
+
+		[Export ("lookupTableForUInt8Kernel")]
+		IntPtr /* float* */ LookupTableForUInt8Kernel { get; }
+	}
+
+	interface IMPSNnPadding { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject), Name = "MPSNNPadding")]
+	interface MPSNnPadding : NSSecureCoding
+	{
+		[Abstract]
+		[Export ("paddingMethod")]
+		MPSNnPaddingMethod PaddingMethod { get; }
+
+		[Export ("label")]
+		string Label { get; }
+
+		[Export ("destinationImageDescriptorForSourceImages:sourceStates:forKernel:suggestedDescriptor:")]
+		MPSImageDescriptor SourceStates (MPSImage[] sourceImages, [NullAllowed] MPSState[] sourceStates, MPSKernel kernel, MPSImageDescriptor inDescriptor);
+	}
+
+	interface IMPSImageSizeEncodingState { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MPSImageSizeEncodingState
+	{
+		[Abstract]
+		[Export ("sourceWidth")]
+		nuint SourceWidth { get; }
+
+		[Abstract]
+		[Export ("sourceHeight")]
+		nuint SourceHeight { get; }
+	}
+
+	interface IMPSImageAllocator { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MPSImageAllocator : NSSecureCoding
+	{
+	    [Abstract]
+	    [Export ("imageForCommandBuffer:imageDescriptor:kernel:")]
+	    MPSImage GetImage (IMTLCommandBuffer cmdBuf, MPSImageDescriptor descriptor, MPSKernel kernel);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSNnStateNode), Name = "MPSCNNConvolutionStateNode")]
+	interface MPSCnnConvolutionStateNode
+	{
+	}
+
+	interface IMPSImageTransformProvider { }
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MPSImageTransformProvider : NSCoding
+	{
+	    [Abstract]
+	    [Export ("transformForSourceImage:handle:")]
+	    MPSScaleTransform GetTransform (MPSImage image, [NullAllowed] IMPSHandle handle);
+	}
+
+	[TV (11, 0), Mac (10,13, onlyOn64: true), iOS (11, 0)]
+	[Protocol]
+	interface MPSDeviceProvider
+	{
+	    [Abstract]
+	    [Export ("mpsMTLDevice")]
+	    IMTLDevice MpsMTLDevice { get; }
```

Comment:
That's not a property in ObjC/swift and might be costly to call.
Switch this to a method, with a better name, like `GetMTLDevice ()`